formats/json/build.gradle

```diff
@@ -26,6 +26,16 @@ kotlin {
                 api project(":kotlinx-serialization-core")
             }
         }
+        jsMain {
```

Comment:
Why `jsMain` has to be changed?

Reply:
Because you moved some files to share them between WASM and JS? Is KMM plugin OK with that? Does import in IDE work correctly?

Reply:
I do not see any issues to do so. But this commit is not necessary and we can drop it (but we need to copy-paste this files into different targets). Or we can try make new source set and depend on it from target source sets, but then we need to ignore metadata tasks for it.

Reply:
Its a bug in IDE. The only thing we can do is copy-paste the code. And this is only inspection bug (the gutter works fine).

Reply:
I wanna see this IDE bug fixed :) So please report it, if it is not created yet.
IDE consistency and correctness directly affects daily work, so yes, I do not want to see any red code in the IDE if it can be compiled correctly. Alternatively, it is possible to use `@Suppress("NO_ACTUAL_FOR_EXPECT")` but I'm not sure that suppressing errors is better than copy pasting several files. @qwwdfsad Maybe you have an opinion on this?

Reply:
I have found that in the latest version actuals in a shared module do work, but they require correct dependencies (the shared module must depend on the common module). A case that still breaks is where you want to share code between jvm and android and use (shared) jvm types.

Reply:
I see that this invalid report occurs on current `dev` branch (without wasm target added), so it seems it is not a result of my changes.

Reply:
https://youtrack.jetbrains.com/issue/KTIJ-26668/False-positive-diagnostic-for-missing-actual-declararion-for-expect-object

Reply:
So what do you prefer?

- Copy and Paste (does not work anyway, the diagnostic will appear)
- Directory sharing
- Sourceset dependency

Reply:
I'm on `dev` and I don't see this error. Did you reimport project after branch switching?

Reply:
I made new project.

Reply:
How it is related to serializtion then? With only jsMain, this error doesn't happen.

We'll discuss it on the next week, since now it is time to release 1.6.0 (this PR won't go there anyway)

Reply:
Well, as I see, with my IDEA it does not related at all.
And I see that with any other MPP project (even without wasm).
I dont understand why it so bother you, its a bug and it is reported.

<img width="725" alt="Screenshot 2023-08-17 at 15 55 17" src="https://github.com/Kotlin/kotlinx.serialization/assets/54274820/28c588e9-6f02-446f-8a6e-8858fb70aea2">

Reply:
I am trying understand the reasons for all of that....if tomorrow new bugs will appear in IDEA, would you rewrite your code to escape this bug or what? Or stop to getting MR's that affected with this bug in editor? What is going on? Why I am wasting my (and yours time). If you are so aware for editor problems with green code, push the IDEA but not codeowners.

Reply:
It's not a stopper for the PR, indeed. I just want to find a suitable workaround.

Reply:
Well, I discussed this with a guys from MPP and we seems to be found whats wrong. Its `.idea` directory that contains artefacts that infer with compiler settings. `.idea` directory does not deleted by any recoveries or cache cleaning options of IDEA, that is why diagnostics persists. You can remove `.idea` folder and this error should away.
We can also compare old `.idea` files and new ones and found the cause (i have already deleted this directory so cannot compare it.

Reply:
Great find, thanks! I'll take a look at .idea

Reply:
I don't know what changed exactly, but the error seems gone for me after switching branches. I guess we can mark it as resolved for now, applying `@Suppress` if it happens again. As I said, I'll merge this PR after 1.6.0 release (tomorrow)

Reply:
Perfect. Thanks. Is it ok for you current approach with intermediate `sourceSet` and ignoring `compileJsWasmMainKotlinMetadata` task as I do for now?

Reply:
> ignoring compileJsWasmMainKotlinMetadata task

Can you elaborate on that? I don't see any changes in buildfiles relating to this task